src/systems/fem_system.C

```diff
@@ -871,9 +871,20 @@ void FEMSystem::assembly (bool get_residual, bool get_jacobian)
                                          mesh.active_local_elements_end()),
                         AssemblyContributions(*this, get_residual, get_jacobian));
 
+  // Check and see if we have SCALAR variables
+  bool have_scalar = false;
+  for(unsigned int i=0; i != this->n_vars(); ++i)
```

Comment:
Good call, thanks for the point that out. I'll push that update in a moment (after running GRINS tests).